src/service/message.js

```diff
@@ -24,62 +49,89 @@ const findAll = () => {
   return [];
 };
 
-const findOne = id => {
+const findOne = messageId => new Promise((resolve, reject) => {
   if (fs.existsSync('log.txt')) {
-    const fileData = fs.readFileSync('log.txt', 'utf8');
+    const rl = createInterface();
 
-    if (fileData) {
-      const message = fileData.split('\n')[id];
+    rl.on('line', line => {
+      if (line) {
+        const { id, message } = JSON.parse(line);
 
-      if (message) {
-        return message;
+        if (id === parseInt(messageId, 10)) {
+          resolve(message);
+          rl.close();
+          rl.removeAllListeners();
+        }
       }
-    }
-  }
+    });
 
-  return null;
-};
+    rl.on('close', () => {
+      reject(new Error('Message has not been found'));
+    });
+  } else {
+    reject(new Error('file not exiest'));
+  }
+});
 
+const deleteMessage = messageId => new Promise((resolve, reject) => {
+  let lineBreak = '';
 
-const deleteMessage = id => {
   if (fs.existsSync('log.txt')) {
-    const fileData = fs.readFileSync('log.txt', 'utf8');
-
-    if (fileData) {
-      const messages = fileData.split('\n');
-
-      if (id >= messages.length) { throw new Error('Message has not been found'); }
-
-      messages.splice(id, 1);
-
-      fs.writeFileSync('log.txt', messages.join('\n'));
-
-      return;
-    }
+    const rl = createInterface();
+
+    rl.on('line', line => {
+      if (line) {
+        const { id, message } = JSON.parse(line);
+
+        if (id !== parseInt(messageId, 10)) {
+          fs.appendFileSync('log1.txt', `${lineBreak}${JSON.stringify({ id, message })}`);
+        } else {
+          resolve();
```

Comment:
You should call `resolve` only when file is ready and closed. Currently it works like:
1. reading file started
2. first line copied to `log1.txt`
3. second line skipped
4. promise resolved (user can continue working)
5. other lines are copying to `log1.txt`
6. `log.txt` is removed
7. `log1.txt` is renamed